implementations/rust/ockam/ockam_transport_tcp/src/workers/sender.rs

```diff
@@ -114,6 +118,30 @@ impl TcpSendWorker {
 
         Ok(())
     }
+
+    fn serialize_message(&self, local_message: LocalMessage) -> Result<Vec<u8>> {
+        // Create a message buffer with prepended length
+        let transport_message = TcpTransportMessage::from(local_message);
+
+        let msg_len = minicbor::len(&transport_message);
+
+        if msg_len > MAX_MESSAGE_SIZE {
+            return Err(TransportError::MessageLengthExceeded)?;
+        }
+
+        let len = 4 /* len */ + msg_len;
```

Comment:
Maybe instead of `/* len */` a more explicit comment saying `We need 4 bytes to serialize the size of the message`.